### ISSUE TITLE
Support specifying arbitrary subject in CSR

### ISSUE THREAD

**Issue by elliotkendall** [ly4k/Certipy#44]
In a recent test, I discovered a template with EnrolleeSuppliesSubject as its Certificate Name Flag. In theory, that means I should be able to specify any subject I like in the CSR and get back a certificate to match. It doesn't seem like certipy supports doing that, though. The req command supports "-alt" to specify a subjectAlternateName, but that's not what we need here.

I may try to write a patch to add this support. Any suggestions on how to go about that in keeping with the style of the project would be welcome. Maybe a new -subject option to req? Maybe support providing your own CSR?

**Comment by ly4k** [comment]
The -alt flag is indeed used to specify the SAN when the EnrolleeSuppliesSubject flag is set, so this feature is already implemented. We can already control the subject for the certificate in most cases, but the subject is not used during authentication - only the SAN

**Comment by elliotkendall** [comment]
Makes sense, thanks. In this particular case I'm not able to specify a SAN - the cert I get back has "X509v3 Subject Alternative Name: othername:<unsupported>". But I think that just means it's not vulnerable.
